From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Thu, 20 Aug 2020 21:01:20 +0200
Subject: [PATCH] {BUG in 5.9-rc1} drivers/block/rnbd/rnbd-srv: initialize ret

clang on x86-64 reports:

    drivers/block/rnbd/rnbd-srv.c:150:6: error: variable 'err' is used
    uninitialized whenever 'if' condition is true
    [-Werror,-Wsometimes-uninitialized]
            if (IS_ERR(bio)) {
                ^~~~~~~~~~~
    drivers/block/rnbd/rnbd-srv.c:177:9: note: uninitialized use occurs
    here
            return err;
                   ^~~
    drivers/block/rnbd/rnbd-srv.c:150:2: note: remove the 'if' if its
    condition is always false
            if (IS_ERR(bio)) {
            ^~~~~~~~~~~~~~~~~~
    drivers/block/rnbd/rnbd-srv.c:126:9: note: initialize the variable
    'err' to silence this warning
            int err;
                   ^
                    = 0
---
 drivers/block/rnbd/rnbd-srv.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/block/rnbd/rnbd-srv.c b/drivers/block/rnbd/rnbd-srv.c
index 0fb94843a495..5b69bc56b225 100644
--- a/drivers/block/rnbd/rnbd-srv.c
+++ b/drivers/block/rnbd/rnbd-srv.c
@@ -149,6 +149,7 @@ static int process_rdma(struct rtrs_srv *sess,
 	bio = rnbd_bio_map_kern(data, sess_dev->rnbd_dev->ibd_bio_set, datalen, GFP_KERNEL);
 	if (IS_ERR(bio)) {
 		rnbd_srv_err(sess_dev, "Failed to generate bio, err: %ld\n", PTR_ERR(bio));
+		err = PTR_ERR(bio);
 		goto sess_dev_put;
 	}
 
-- 
